src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1404,6 +1417,16 @@ def _handle_sql_queue_item(self, item):
     def _update_failover_status(self):
         self.failover_status = self.middleware.call_sync('failover.status')
 
+    def _os_versions_match(self):
+
+        try:
+            loc = self.middleware.call_sync('failover.get_os_version')
+            rem = self.middleware.call_sync('failover.call_remote', 'failover.get_os_version')
```

Comment:
I am confused, get_os_version already gets the version of the remote, no?

Reply:
No it just gets the cached version that is set in `RemoteClient`.

Reply:
What is cached on RemoteClient is the remove version. So we are calling the remote node get its cached version of the other remote node (us?)?

Reply:
No, the `RemoteClient` is running `self.client.call('system.version')` which is the local OS version and then setting the `self._os_version` attribute.

Reply:
The `JournalSync` thread is running `failover.get_os_version` which gets the `RemoteClient`'s cached `self._os_version` attribute.

Reply:
`self.client.call('system.version')` definitely doesnt get the local version.